atlasdb-api/src/main/java/com/palantir/atlasdb/persistent/api/LogicalPersistentStore.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.persistent.api;
+
+import java.util.List;
+import java.util.Map;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import com.palantir.atlasdb.persistent.api.PersistentStore.StoreNamespace;
+
+public interface LogicalPersistentStore<K, V> {
+    /**
+     * Retrieve the value associated with the given {@code key}.
+     *
+     * @param storeNamespace from which to retrieve the value
+     * @param key            of the cache entry
+     * @return value associated or null if the entry is missing
+     */
+    @Nullable
+    V get(StoreNamespace storeNamespace, @Nonnull K key);
+
+    /**
+     * Retreives values for all supplied {@code keys}.
+     *
+     * @param storeNamespace from which to retrieve the values
+     * @param keys           for which entries we are interested in
+     * @return map of key, value pairs containing only entries which are stored
+     */
+    Map<K, V> multiGet(StoreNamespace storeNamespace, List<K> keys);
+
+    /**
+     * Stores the given entry pair.
+     *
+     * @param storeNamespace where to store the entry
+     * @param key            of the entry
+     * @param value          of the entry
+     */
+    void put(StoreNamespace storeNamespace, @Nonnull K key, @Nonnull V value);
+
+    /**
+     * Stores all entry pairs specified in {@code toWrite}.
+     *
+     * @param storeNamespace where to store the entries
+     * @param toWrite        entry pairs to store
+     */
+    void multiPut(StoreNamespace storeNamespace, Map<K, V> toWrite);
+
+    /**
+     * Creates a {@link StoreNamespace} with the given name. Multiple calls with the same {@code name} will return
+     * different namespaces.
+     *
+     * @param name of the namespace
+     * @return handle to the underlying namespace
+     */
+    StoreNamespace createNamespace(@Nonnull String name);
```

Comment:
The name is misleading and unintuitive and not really comparable to a `TableReference`. It should be called something else.

Reply:
also, I'd drop the concept of namespace, it's currently overloaded i.e. it's not the same thing as a timelock namespace or a c* namespace/keyspace and it's an extra concept. Something along the lines of: `createHandleForNamespace` if that's the distinction you're going for.

Reply:
Agree, though I think we should track this on another PR (`StoreNamespace` is already used quite a lot through this project).